src/fsharp/FSharp.Core/Linq.fs

```diff
@@ -355,141 +427,107 @@ module LeafExpressionConverter =
                 let props = ctor.DeclaringType.GetProperties()
                 Expression.New(ctor, argsR, [| for p in props -> (p :> MemberInfo) |]) |> asExpr
 
-
             // Do the same thing as C# compiler for string addition
-            | PlusQ (_, [ty1; ty2; ty3], [x1; x2]) when (ty1 = typeof<string>) && (ty2 = typeof<string>) && (ty3 = typeof<string>) ->
+            | PlusQ (_, m, [x1; x2]) when m.GetGenericArguments() = arrayOf3StringTypes ->
```

Comment:
I'd still prefer not to see array equality being used in this code.  Also `GetGenericArguments` can fail if not `IsGenericMethod` - please double check we have a guarantee from `PlusQ` that this is a generic method and if so add a comment here to that effect?

Reply:
Why no array equality?

Reply:
It has unnecessary performance overheads for this kind of code (at least one type test is needed).